### PR TITLE
Add retry loop in generated client for idempotent operations

### LINKED ISSUE #19: Idempotent IPC doesn't actually retry
It's implied in the docs and comments that functions marked as idempotent will retry, and that therefore we don't need to worry about server death.

Retrying doesn't actually happen!

The actual behavior depends on return mode:
- `Simple` will ignore dead codes and try to unpack regardless
- `Result` will call `from_u32` on the error type, which will fail unless that type can decode dead codes (which nothing in Hubris does)

Either of these seems bad!

### PR DESCRIPTION
**Pull request by jgallagher**
It looked like the simplest way to conditionally introduce the loop was to change the final statement to always be (the equivalent of) `return v;` instead of just `v`, so I also added a suppression for `clippy::needless_return`, since all non-idempotent operations now end with a needless return.

Fixes #19.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -80,12 +81,27 @@ pub fn generate_client_stub(
             if op.leases.values().any(|lease| lease.read && lease.write) {
                 return Err("idempotent operation with read/write lease".into());
             }
+            match &op.reply {
+                syntax::Reply::Result { err, .. }
+                    if matches!(err, syntax::Error::ServerDeath) =>
+                {
+                    return Err(
+                        format!("idempotent operations should not indicate server death: {}", name)
```
Comment (cbiffle) [inline src/client.rs:89]: nitty nit nit:

```suggestion
                        format!("idempotent operations should not indicate server death: {name}")
```

File: src/client.rs [hunk 2]
```diff
@@ -80,12 +81,27 @@ pub fn generate_client_stub(
...
+                _ => (),
+            }
         } else {
             // A non-idempotent operation had better have an error type.
             match &op.reply {
                 syntax::Reply::Result { .. } => (),
                 syntax::Reply::Simple(_) => {
-                    return Err("operation can't indicate server death".into());
+                    return Err(format!(
+                        "operation can't indicate server death: {}",
```
Comment (cbiffle) [inline src/client.rs:101]: (could also use `{name}`  here)

**Review by cbiffle — APPROVED** [review]
And thanks @mkeeter for noticing that I _totally forgot to do this._

File: src/client.rs (hunk 1 above)
Comment (jgallagher) [reply src/client.rs:89]: That doesn't work with the `rust-toolchain.toml` file in this repo :grimacing:

Comment (jgallagher) [reply src/client.rs]: `rust-toolchain.toml` updated to match hubris

**Pull request merged**